### PR TITLE
[GetAndroidDependencies] Add Jdk dependency info

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/xamarin-android-tools/commit/ed102fc90870f87b958fec82414cf2dc909e15c2

Updates `<GetAndroidDependencies/>` to output a new `@(JavaDependency)`
item group if `$(AndroidInstallJavaDependencies)` is set to `true`.
This group currently contains one item:

    JavaDependency
      jdk
        Version = 17.0.8.1

This output will be used by `<InstallAndroidDependencies/>` to install
the our recommended version of the Java SDK.

### CODE REVIEW

**Review by jonathanpeppers — COMMENTED** [review]
Is there way to update an existing test to check this works?

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 1]
```diff
@@ -2731,8 +2731,10 @@ because xbuild doesn't support framework reference assemblies.
     PlatformToolsVersion="$(AndroidSdkPlatformToolsVersion)"
     NdkVersion="$(AndroidNdkVersion)"
     NdkRequired="$(_NdkRequired)"
+    JdkVersion="$(JavaSdkVersion)"
   >
     <Output TaskParameter="Dependencies" ItemName="AndroidDependency" />
+    <Output TaskParameter="JavaDependencies" ItemName="JavaDependency" />
```
Comment (pjcollins) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2737]: Looping in @tondat to help review with respect to the IDE and SDK Manager. Using a new `@(JavaDependency)` output item should hopefully reduce the chance of potential breakage in other areas that depend on `GetAndroidDependencies`

**Comment by pjcollins** [comment]
> Is there way to update an existing test to check this works?

I tried to update an existing test to check for the new `JavaDependency` output, hopefully that will work. I was also planning on adding tests for installation once https://github.com/xamarin/android-sdk-installer/pull/913 makes its way through the dependency chain.

**Review by dellis1972 — APPROVED** [review]

**Review by jonathanpeppers — APPROVED** [review]

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets (hunk 1 above)
Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2737]: We found [a couple of places](https://github.com/xamarin/XamarinVS/blob/313d332b37180718cbf77864d2afbc38b8c027bc/src/Core/VisualStudio.Android/Services/AndroidToolsBuildService.cs#L87) where the return value seems to be used without name / itemspec filtering, so I am going to try to remove `@(JavaDependency)` from the target return value for now.

Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2737]: @dellis1972 @jonathanpeppers given the potential breakage with IDE/SDK Manager, I am leaning towards adding a separate `GetJavaDependencies` target that [`<InstallAndroidDependencies/>`](https://github.com/xamarin/android-sdk-installer/blob/68939f222b0e794424123ce6cb46bf0ba7ef73a4/Xamarin.Installer.Build.Tasks/Xamarin.Installer.Common.targets#L11) can rely on, does that approach seem reasonable?

Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2737]: To keep the changes smaller, you could also do a new MSBuild property? `-p:AndroidReturnJavaDependencies=true`

The existing `InstallAndroidDependencies` target would set this, but the existing IDEs wouldn't:

```xml
<Target Name="_SetAndroidReturnJavaDependencies">
  <PropertyGroup>
    <AndroidReturnJavaDependencies Condition=" '$(AndroidReturnJavaDependencies)' == '' ">true</AndroidReturnJavaDependencies>
  </PropertyGroup>
</Target>

<Target Name="InstallAndroidDependencies" DependsOnTargets="_SetAndroidReturnJavaDependencies;GetAndroidDependencies">
<!-- ... -->
```
Then your `<CalculateProjectDependencies/>` task would only return Java stuff if `$(AndroidReturnJavaDependencies)` was true.

I think either option would work, it's either invent a new property or new target?

Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2739]: Added a property to opt out of this new item by default which should hopefully address the concerns of external dependencies.

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 2]
```diff
@@ -2712,6 +2712,7 @@ because xbuild doesn't support framework reference assemblies.
 <Target Name="_BeforeGetAndroidDependencies">
   <PropertyGroup>
     <_AndroidAllowMissingSdkTooling>True</_AndroidAllowMissingSdkTooling>
+    <AndroidGetJavaDependencies Condition=" '$(AndroidGetJavaDependencies)' == '' ">false</AndroidGetJavaDependencies>
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:2715]: Looks good, but we also need to document the public property in the build-properties.md now :)

**Review by jonathanpeppers — APPROVED** [review]

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets (hunk 2 above)
Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets]: Thanks, I've updated the property docs

**Pull request merged**